### DISCUSSION TITLE
Cell vectors changing in NVT ensemble

### DISCUSSION THREAD

**Discussion by achyuttt** [dftbplus/dftbplus#1489] [General]
Hello everyone, 

I am performing NVT MD in DFTB. Despite the md.out file has the same volume for each steps the visualization of .xyz file in ovito indicates that the cell parameters are changing as per the MD time steps.

My input file looks like this 👍 

Can anyone suggest me whats wrong in the input file or if there's something that I can change in the input file? so that I can get the MD simulation of my system intact in NVT ensemble.

Geometry = GenFormat {
<<< "geo_400.gen"
}
Driver = VelocityVerlet {
  MovedAtoms = 1:-1
  Steps = 2000
  TimeStep [fs] = 2
  Thermostat = NoseHoover {
    Temperature [Kelvin] = 400.0
    AdaptFillingTemp = Yes
    CouplingStrength [cm^-1] = 3250
    ChainLength = 5
    Order = 3
    IntegratorSteps = 1
  }
  OutputPrefix = "md"
  MDRestartFrequency = 1
  Velocities [AA/ps] = {
  <<< "velocity_400.gen"
  }
  Masses = {}
  KeepStationary = Yes
  Plumed = No
}
Hamiltonian = DFTB {
  SCC = Yes
  SCCTolerance = 1e-6
  MaxSCCIterations = 100000
  SlaterKosterFiles = Type2FileNames {
    Prefix = ".../Potential/skfiles/mio-1-1/"
    Separator = "-"
    Suffix = ".skf"
    LowerCaseTypeName = No
  }
  MaxAngularMomentum = {
    C = "p"
    H = "s"
    N = "p"
  }
  KPointsAndWeights = SupercellFolding {
3 0 0
0 3 0
0 0 1
0.5 0.5 0.5
  }
  ForceEvaluation = Dynamics
  Filling = Fermi {
    IndependentKFilling = No
  }
  PolynomialRepulsive = {}
  ShellResolvedSCC = No
  OldSKInterpolation = No
  ReadInitialCharges = No
  InitialCharges = {}
  EwaldParameter = 0.0000000000000000
  EwaldTolerance = 1.0000000000000001E-009
  ConvergentSCCOnly = Yes
  HCorrection = None {}
  SpinPolarisation = {}
  ElectricField = {}
  Solver = RelativelyRobust {}
  Charge = 0.0000000000000000
  OnSiteCorrection = {}
  Dispersion = LennardJones{
	Parameters = UFFParameters{}
  }
  Solvation = {}
  Electrostatics = GammaFunctional {}
  ThirdOrder = No
  ThirdOrderFull = No
  Differentiation = FiniteDiff {
    Delta = 1.2207031250000000E-004
  }
  Mixer = Broyden {
    MixingParameter = 0.20000000000000001
    InverseJacobiWeight = 1.0000000000000000E-002
    MinimalWeight = 1.0000000000000000
    MaximalWeight = 100000.00000000000
    WeightFactor = 1.0000000000000000E-002
  }
}
Options = {
  RandomSeed = 3871906
  WriteResultsTag = Yes
  WriteDetailedOut = No
  WriteAutotestTag = No
  WriteDetailedXML = Yes
  WriteHS = No
  WriteRealHS = No
  MinimiseMemoryUsage = No
  ShowFoldedCoords = No
  TimingVerbosity = 1
  WriteCharges = Yes
  WriteChargesAsText = No
  BinaryAccessTypes = {
"stream"
  }
}
Analysis = {
  Printforces = Yes
  ProjectStates = {}
  WriteEigenvectors = No
  WriteBandOut = No
  MullikenAnalysis = Yes
  WriteNetCharges = No
  AtomResolvedEnergies = No
}
ParserOptions = {
  ParserVersion = 14
  WriteHSDInput = Yes
  StopAfterParsing = No
  IgnoreUnprocessedNodes = No
}
Parallel = {
  UseOmpThreads = Yes
  Groups = 1
  Blacs = BlockSize {
32
  }
}
Reks = None {}
ExcitedState = {}

**Comment by rvipa-dtu** [comment]
hi! did you ever figure this out? i'm trying to run nvts as well and i have some strange behavior too
